### PR TITLE
#16851 Filings UI: Styling issues in the correction module

### PR DESCRIPTION
**Pull request by bennypc**
*Issue #:* /bcgov/entity#16851

*Description of changes:*

Made styling and functional changes based on the ones listed in the ticket here: https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/16851

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/Certify.vue
Comment (severinbeauvais) [inline src/components/common/Certify.vue:129]: Adding this class is fine (since you can't reuse "certify-clause" because it's used for other elements), but I recommend a more description (and longer) name than "date".

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:596]: This double-deep refs is not good practice. (It's bad enough to reach into 1 component to set or call something.)

I think it's also bad practice to set a component's error to True. Instead, it would be better to assign some "rules" to that component and then having the component check them (eg, by calling `validate()` on it, if needed).

Comment (severinbeauvais) [reply src/views/Correction.vue:596]: The Detail Comment component may also accept a prop to ask it to validate itself.

File: src/views/Correction.vue
Comment (severinbeauvais) [reply src/views/Correction.vue:904]: Same for line 929 below.

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:906]: `placeholder` is not a valid type selector (eg, div, p, li, etc). If it's a class then it need a `.`, or if it's an id then it needs a `#`.

Comment (severinbeauvais) [reply src/views/Correction.vue:906]: Ditto on line 933 below.

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:904]: Double empty lines are causing a build error. Just have 1 blank line.

File: src/views/Correction.vue
Comment (severinbeauvais) [reply src/views/Correction.vue]: Sorry about the confusion -- `:deep` or `::v-deep` are the old syntaxes. The new syntax is `deep()` but it doesn't work in all UIs yet.

Comment (bennypc) [reply src/views/Correction.vue]: so I should not use v-deep and instead just use deep()?

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:935]: I bet you don't need this block, since you already do this on lines 904-906.

Comment (bennypc) [reply src/views/Correction.vue:935]: oh i named the selector wrong good catch, its for detail-comment-input

File: src/views/Correction.vue
Comment (severinbeauvais) [reply src/views/Correction.vue]: Whatever the current version of Vue/Vuetify/compiler? supports. See surrounding code.

Eventually I think it will all be `deep()` but that doesn't work yet in some UIs.

**Comment by severinbeauvais** [comment]
You don't need to re-request a review from me unless I've totally forgotten about this (no response for a day). Otherwise, I get notifications when you respond to a conversation or commit new code, and then I'll go look at the changes.

**Comment by severinbeauvais** [comment]
I still have unresolved conversations above. If you disagree with my comment then let's have a chat about it 😄

**Comment by bennypc** [comment]
> I still have unresolved conversations above. If you disagree with my comments then let's have a chat about it 😄

Just wanted to check the deep() issue first, currently working on the other ones, sorry!

File: src/views/Correction.vue
Comment (JazzarKarim) [reply src/views/Correction.vue]: Were you able to implement this?

**Comment by pwei1018** [comment]
Temporary Url for review: https://business-filings-dev--pr-529-abbcrdri.web.app

File: src/views/Correction.vue
Comment (chenhongjing) [inline src/views/Correction.vue:87]: There are corrections for different filings. So you should replace `Record Conversion` with its corresponding type. 
For example, the following subtitle should be "Correction for Dissolution"
image

File: src/views/Correction.vue
Comment (JazzarKarim) [inline src/views/Correction.vue:605]: Please move this comment to line 608.

File: src/views/Correction.vue
Comment (bennypc) [reply src/views/Correction.vue:87]: is there a type i can use for this?

File: src/views/Correction.vue
Comment (JazzarKarim) [reply src/views/Correction.vue]: Fixed. Resolving.

File: src/views/Correction.vue
Comment (JazzarKarim) [reply src/views/Correction.vue]: Fixed. Resolving.

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:106]: You probably don't need the div around this -- you can add the class directly to this component itself, like any other HTML element.

Ditto line 132 below.

File: src/views/Correction.vue
Comment (severinbeauvais) [inline src/views/Correction.vue:873]: Excellent use of the existing mechanism to implement auto scrolling in this component!
👍👍

File: src/components/common/Certify.vue
Comment (severinbeauvais) [inline src/components/common/Certify.vue:60]: Please rename to `checkboxRef`.

File: src/components/common/Certify.vue
Comment (severinbeauvais) [inline src/components/common/Certify.vue:82]: Did you run `npm run lint`? Did it complain about the trailing semi-colon?

File: src/views/Correction.vue
Comment (severinbeauvais) [reply src/views/Correction.vue]: This code was re-written and this conversation is obsolete. Resolving.

File: src/views/Correction.vue
Comment (severinbeauvais) [reply src/views/Correction.vue]: Haha, I see there was already a variable you could use... nice :)

**Review by severinbeauvais — APPROVED** [review]
Just a couple of minor things to fix up and then, after another approval or 2, you're good to merge.

**Review by JazzarKarim — APPROVED** [review]
LGTM but please make sure to fix outstanding comments 👍

File: src/components/common/Certify.vue
Comment (bennypc) [reply src/components/common/Certify.vue:82]: I did, it didn't make any changes there but I can remove it

Comment (severinbeauvais) [reply src/components/common/Certify.vue]: Yes, please. Thanks!

Comment (bennypc) [reply src/components/common/Certify.vue]: Did it! Should be in the new commit

**Pull request merged**
